src/Win32Util.cpp

```diff
@@ -67,9 +69,11 @@ argv_to_string(const char* const* argv, const std::string& prefix)
     for (size_t j = 0; arg[j]; ++j) {
       switch (arg[j]) {
       case '\\':
-        ++bs;
-        break;
-      // Fallthrough.
+        if (!escape_backslashes) {
+          ++bs;
+          break;
+        }
+        // Fallthrough.
       case '"':
         bs = (bs << 1) + 1;
```

Comment:
@jrosdahl Could you explain the logic of `bs` variable? I made the change, it works for me, but the be honest I don't understand well why `bs` is manipulated in such complex way.

Reply:
> Could you explain the logic of `bs` variable?

No, sorry.

> What is the reasoning behind it?

I have no idea since it wasn't written by me. It was added in 4d7b4810bd3b8b53832605927c100a5ba4a60fb1 with a reference to <http://msdn.microsoft.com/en-us/library/17w5ykft.aspx>. The Windows parts of ccache are in practice maintained by ccache community members who care about Windows, now including you. :slightly_smiling_face:

Reply:
Thanks for MSDN reference. I'll keep it for further analysis in hypothetical free time :-)